src/Controller/ProposalController.php

```diff
@@ -1,86 +1,134 @@
-        $this->getEntityManager()->remove($entity);
+        if ($proposition === $word) {
+            $word_status = $proposition;
+        } elseif (strlen($proposition) === 1) {
+            $propositionLetter = $proposition[0];
+            $word_statusArr = str_split($word_status);
+
+            foreach ($word_statusArr as $index => $letter) {
+                if ($letter === '_' && $word[$index] === $propositionLetter) {
+                    $word_statusArr[$index] = $propositionLetter;
+                }
+            }
 
-        if ($flush) {
-            $this->getEntityManager()->flush();
+            $word_status = implode('', $word_statusArr);
         }
+        return $word_status;
     }
 
-    public function findByGame(int $gameId): array
+    /**
+     * Get the points won
+     * @param string $word
+     * @param string $word_status
+     * @param string $proposition
+     * @return int
+     */
+    private function getPoints(string $word, string $word_status, string $proposition): int
     {
-        return $this->createQueryBuilder('p')
-            ->andWhere('p.game = :gameId')
-            ->setParameter('gameId', $gameId)
-            ->getQuery()
-            ->getResult();
-    }
+        $wordArr = str_split($word);
+        $wordStatusArr = str_split($word_status);
+        $propositionArr = str_split($proposition);
 
-    public function findByUser(int $userId): array
-    {
-        return $this->createQueryBuilder('p')
-            ->andWhere('p.user = :userId')
-            ->setParameter('userId', $userId)
-            ->getQuery()
-            ->getResult();
-    }
+        $discoveredLetters = [];
 
-    public function findByUserAndGame(int $userId, int $gameId)
-    {
-        try {
-            return $this->createQueryBuilder('p')
-                ->andWhere('p.user = :userId')
-                ->andWhere('p.game = :gameId')
-                ->setParameter('userId', $userId)
-                ->setParameter('gameId', $gameId)
-                ->getQuery()
-                ->getOneOrNullResult();
-        } catch (NonUniqueResultException $e) {
-            return null;
-        }
-    }
+        if (count($propositionArr) === 1) {
+            $letter = $propositionArr[0];
+            $index = array_search($letter, $wordStatusArr);
 
-    public function findCorrectByGame(int $gameId): array
-    {
-        return $this->createQueryBuilder('p')
-            ->andWhere('p.game = :gameId')
-            ->andWhere('p.is_correct = true')
-            ->setParameter('gameId', $gameId)
-            ->getQuery()
-            ->getResult();
+            if ($index !== false && $wordStatusArr[$index] === '_') {
+                $discoveredLetters[] = $letter;
+            }
+        } else {
+            if ($proposition === $word) {
+                foreach ($wordArr as $index => $letter) {
+                    if ($wordStatusArr[$index] === '_') {
+                        $discoveredLetters[] = $letter;
```

Comment:
Should we add a condition to say if the letter is not already included in the discovered letters ?